authors/apps/friends/views.py

```diff
@@ -0,0 +1,130 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from django.shortcuts import get_object_or_404
+
+from authors.apps.authentication.models import User
+from authors.apps.friends.models import Friend
+from authors.apps.friends.serializers import (
+    FollowerFollowingSerializer, FollowUnfollowSerializer
+)
+from authors.apps.friends.utils import Utils
+
+
+class FollowUnfollowAPIView(generics.RetrieveUpdateDestroyAPIView):
+    """
+    APIView for following a user
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, username, format=None):
+        """
+        This method create a user relationship btween the user 
+        with the username passed in and the user sending the 
+        request
+        """
+        try:
+            to_be_followed = User.objects.get(username=username)
+        except Exception:
+            return Response({
```

Comment:
Hell Paul. Because its a `resource that can not be found on the` **url**, you could raise a `NotFound` Exception instead of returning a Response. Check it out here
https://www.django-rest-framework.org/api-guide/exceptions/

Reply:
On it

Reply:
@codact I have researched and settled on presenting custom error messages that non-technical users can understand.

Reply:
ok..that will still work fine